lib/plugins/posthtml-flatten-image.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+var url = require('url');
+var Bluebird = require('bluebird');
+var collapseBinary = require('../collapsers/binary');
+var base64Utils = require('../utils/base-64');
+
+module.exports = function (opts) {
+  return function posthtmlFlattenImage(tree) {
+    return Bluebird.try(function () {
+      var tasks = [];
+
+      tree.match({
+        tag: 'img',
```

Comment:
Probably let's keep in mind and add in the separate PR then, since irrelevant to the refactoring.